cli/src/core/parse/packageSpec_test.go

```diff
@@ -0,0 +1,194 @@
+package parse
+
+import (
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"testing"
+
+	"cli/cmd/flags"
+	"cli/core"
+	"cli/core/state"
+
+	"github.com/luno/jettison/jtest"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func copyFile(src, dst string) error {
+	// Open the source file
+	sf, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer sf.Close()
+
+	// Create the destination file
+	df, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer df.Close()
+
+	// Copy the contents of the source file to the destination file
+	_, err = io.Copy(df, sf)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func TestGetPackageSpecFromParams(t *testing.T) {
+	wd, err := os.Getwd()
+	jtest.RequireNil(t, err)
+
+	type cases struct {
+		configFilePath string
+		hookFunc       func(cmd *cobra.Command)
+		wantSpecMatch  bool
+		packageSpec    *core.PackageSpec
+		errorString    string
+	}
+
+	testCases := []cases{
+		// case: match packageSpec
+		{
+			configFilePath: wd + "/../../features/unit-test-configs/config-case-3.yml",
+			hookFunc: func(cmd *cobra.Command) {
+				cmd.Flags().StringSlice("env-file", []string{""}, "")
+
+				cmd.Flags().Set("name", "pack-1")
+				cmd.Flags().Set("name", "pack-2")
+
+				cmd.Flags().Set("env-file", wd+"/../../features/test-conf/.env.one")
+				cmd.Flags().Set("env-file", wd+"/../../features/test-conf/.env.two")
+
+				cmd.Flags().Set("custom-path", "disi-on-platform")
+
+				cmd.Flags().Set("dev", "true")
+				cmd.Flags().Set("only", "true")
+			},
+			wantSpecMatch: true,
+			packageSpec: &core.PackageSpec{
+				Packages: []string{"pack-1", "pack-2"},
+				CustomPackages: []core.CustomPackage{
+					{
+						Id:   "disi-on-platform",
+						Path: "git@github.com:jembi/disi-on-platform.git",
+					},
+				},
+				EnvironmentVariables: []string{"FIRST_ENV_VAR=number_one", "SECOND_ENV_VAR=number_two"},
+				IsDev:                true,
+				IsOnly:               true,
+			},
+		},
+		// case: return error from not finding env file
+		{
+			configFilePath: wd + "/../../features/unit-test-configs/config-case-3.yml",
```

Comment:
Let's refactor the path inside a var so we don't repeatedly specify.

Reply:
Cool, done in [895254a](https://github.com/openhie/package-starter-kit/pull/34/commits/895254ac521594dec27e0701216486768591e1b7)